program_repair.py

```diff
@@ -1,15 +1,19 @@
+fixed_code = None
+mutants_made = False
+import hedy
+
 def insert(input_string, line, column, new_string):
     """"insert new_string at (line, column)"""
     rows = input_string.splitlines()
-    rows[line] = rows[line][:column] + new_string + rows[line][column:]
+    rows[line - 1] = rows[line - 1][:column - 1] + new_string + rows[line - 1][column - 1:]
```

Comment:
inserting this way is very expensive because split() and join are expensive. Can we rewrite the code so it splits and joins once? So split at the beginning of make_mutants, only pass the one line to insert() and then join all code back in at the end of this? (the is true for your other mutation functions too)

Reply:
I rewrote it such that split() is only done once, but I do not think we can reduce the amount of join().
The amount of join() calls will always be equal to the amount of mutants.